pkg/operator/rules.go

```diff
@@ -110,6 +110,7 @@ func (prs *PrometheusRuleSelector) sanitizePrometheusRulesSpec(promRuleSpec moni
 	minVersionKeepFiringFor := semver.MustParse("2.42.0")
 	minVersionLimits := semver.MustParse("2.31.0")
 	minVersionQueryOffset := semver.MustParse("2.53.0")
+	minVersionRuleGroupLabels := semver.MustParse("3.0.0")
```

Comment:
I don't think that Thanos supports this yet? We need to do the same as for minVersionQueryOffset (see L119 below).